src/gluonts/model/forecast_generator.py

```diff
@@ -114,8 +114,9 @@ def __call__(
         **kwargs
     ) -> Iterator[Forecast]:
         for batch in inference_data_loader:
-            inputs = [batch[k] for k in input_names]
-            outputs = predict_to_numpy(prediction_net, inputs)
+            # TODO: Change inputs to dict to avoid assuming matching order in input_names / model signature?
+            inputs = [batch.get(k) for k in input_names]
+            outputs = prediction_net(*inputs).cpu().detach().numpy()
```

Comment:
My bad, this change is actually unnecessary. 

There is still the question of whether `inputs` should be a `dict` instead of list to remove the implicit assumption that the of arguments in `input_names` matches the order expected by `forward` method of the model.